### PR TITLE
Custom child theme #18

### LINKED ISSUE #6: Build header

### LINKED ISSUE #7: Build footer

### LINKED ISSUE #17: Set up LESS file directory
## Description

Add [Custom Less File](https://github.com/vufind-org/learning-vufind-book/blob/dev/source/08-styling_vufind.rst#832-adding-a-custom-less-file)

Add [Grunt](https://github.com/vufind-org/learning-vufind-book/blob/dev/source/08-styling_vufind.rst#8321-compiling-less-using-javascript)

### LINKED ISSUE #18: Create a CRL custom (child) theme
## Description 

https://vufind.org/wiki/development:architecture:user_interface

### PR DESCRIPTION
**Pull request by KimSarabia**
## Description

- [x] Header style #6 
- [x] Footer style #7
- [x] LESS Files setup #17 
- [x] Custom child theme setup #18 

Miscellaneous Child Theme Components (No tickets)
- [x] Adds favicon
- [x] Adds Fontello icons
- [x] Homepage facets style
- [x] Global style

To-Do

- [x] Mobile/Responsive

closes #6
closes #7
closes #17 
closes #18

### CODE REVIEW

**Comment by KimSarabia** [comment]
Hi @ryan-jacobs This is ready for a cursory aesthetic lookthrough.
The one thing I would appreciate a second paid of eyes for is some compilation behavior:

Screen Shot 2022-04-07 at 2 48 47 PM

Every time I run `grunt less`, it creates two compiled css files instead of just one. It's not a blocker, and it doesn't seem to create a major performance bottleneck, but have not been able to solve this yet. I believe we can merge this even with this issue, but let me know what you think.

File: themes/crl/theme.config.php [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?php
+return [
+    'extends' => 'bootstrap3',
+    'less' => array(
+      'active' => true/false,
+      'compiled.less'
+    ),
```
Comment (ryan-jacobs) [inline themes/crl/theme.config.php:7]: My guess is that these lines are causing a less compilation conflict. It does not look like the other themes that extend bootstrap3 declare these, and even the theme.config.php for booptstrap3 sets `'active' => false` here. Perhaps the `'active' => true/false` line here is getting interpreted as `'active' => true` that that's telling the theme system to do on-the-fly compilation of less files, which then writes out that _extra_ themes/crl/less/compiled.css file on the next page load (if I try to simply delete this themes/crl/less/compiled.css file it just gets regenerated). It also seems tells the system to include a redundant themes/crl/less/compiled.css file in the header, which also creates broken url() references.

I assume the intent here is to compile all css manually in dev environments and then commit the compiled css into VC as opposed to doing on-the-fly compilation in each environment, correct? If so, perhaps the fix is to just remove this 'less' array and also remove the themes/crl/less/ dir from VC.

**Review by ryan-jacobs — COMMENTED** [review]
@KimSarabia , more comments coming soon, but I first wanted to capture a note about the less compilation. I'm not sure if I'm understanding the way the systems is handling things, but I may have uncovered a pointer at least.

**Comment by ryan-jacobs** [comment]
We'll certainly be able to merge this in today, and can track some small adjustments in follow-ups while continuing to style additional components.

I quite like:

- The handling for the search area. As this is a persistent component used across a number of interfaces (e.g. search terms entered on any page get combined with facets from list pages, etc.) having it styled in this unique and bold way globally works well.
- Header menu bar. We need to decide just how we'll use this, and what static content will live here, but I think it's a nice addition and also works well visually.
- Global typography styles are more appropriate and easy to read
- Icons, buttons and pills are much improved.
- Card example from the homepage is very nice (we'll just have to figure out where else to use this style)
- Footer colors and typography (we'll just have to think about which content to keep here... like social links)

Some things for us to consider in discussions during next sprint and as follow-ups:

- How to title the site in the header. We'll probably want to get some "Catalog" title text up there.
- What elements to keep in footer. The main website elements are definitely the right starting point here, I'm just not sure about the social links and sub-site links. This may be one aspect to get some Comm feedback on eventually.
- Possibly styling to break-up some parts of the record page (e.g. should we visually separate the body area from tabbed-sections or do any other container styling here?

**Comment by KimSarabia** [comment]
@ryan-jacobs Ok I'll merge this and make additional tickets for the follow-up section

**Pull request merged**